### PR TITLE
prop types post

### PR DESCRIPTION
**Pull request by EramM805**
PropTypes added to post
Unit tests written for proptypes post

### CODE REVIEW

File: __tests__/CommentsTouchBox-test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import 'react-native';
+import React from 'react';
+import CommentsTouchBox from '@screens/app/Components/CommentsTouchBox/CommentsTouchBox';
+import renderer from 'react-test-renderer';
+import {shallow} from 'enzyme';
+import checkPropTypes from 'check-prop-types';
+var assert = require('assert');
+
+test('like button snapshot test', () => {
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:9]: Comment Button Snapshot test?

File: __tests__/CommentsTouchBox-test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  expect(treeWithCommentLengthZero).toMatchSnapshot();
+  const treeWithCommentLengthOne = renderer
+    .create(<CommentsTouchBox commentsLength={1} />)
+    .toJSON();
+  expect(treeWithCommentLengthOne).toMatchSnapshot();
+  const treeWithCommentLengthGreaterThanOne = renderer
+    .create(<CommentsTouchBox commentsLength={2} />)
+    .toJSON();
+  expect(treeWithCommentLengthGreaterThanOne).toMatchSnapshot();
+});
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:22]: What about snapshot with len < 0? or MAX_INT? Do we have a min/max set? If so, should we create snapshots for those edge cases

File: __tests__/CommentsTouchBox-test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    .toJSON();
+  expect(treeWithCommentLengthZero).toMatchSnapshot();
+  const treeWithCommentLengthOne = renderer
+    .create(<CommentsTouchBox commentsLength={1} />)
+    .toJSON();
+  expect(treeWithCommentLengthOne).toMatchSnapshot();
+  const treeWithCommentLengthGreaterThanOne = renderer
+    .create(<CommentsTouchBox commentsLength={2} />)
+    .toJSON();
+  expect(treeWithCommentLengthGreaterThanOne).toMatchSnapshot();
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:21]: Could we also say "equal to 2?"

File: __tests__/CommentsTouchBox-test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      expect(commentsLength).toEqual(commentsLengthProp);
+    },
+  );
+}
+
+function testPropTypesWithErrors(commentLength, errorMsg, testDescription) {
+  it('Throws failed propType Error for ' + testDescription, () => {
+    let result = checkPropTypes(
+      CommentsTouchBox.propTypes,
+      {commentsLength: commentLength},
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:43]: Generalize this to take any number of props?

File: __tests__/CommentsTouchBox-test.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        <CommentsTouchBox commentsLength={commentsLength} />,
+      );
+      const instance = wrapper.instance();
+      const commentsLengthProp = instance.props.commentsLength;
+      expect(commentsLength).toEqual(commentsLengthProp);
+    },
+  );
+}
+
+function testPropTypesWithErrors(commentLength, errorMsg, testDescription) {
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:39]: I've noticed this function a lot and began to attempt generalizing it on a page ( however we removed that component so I never fully was able to try it out before deleting) but maybe pass in the Component, full props (pre-condition that all other props except the one we want to violate meets the requirements) and then test descriptions?

File: __tests__/CommentsTouchBox-test.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      CommentsTouchBox.commentsLength,
+    );
+    assert(result === errorMsg);
+  });
+}
+
+describe('testing commentsLength values', () => {
+  testPropTypesWithoutErrors(0);
+  testPropTypesWithoutErrors(1);
+  testPropTypesWithoutErrors(1000);
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:54]: What's the max?

File: __tests__/CommentsTouchBox-test.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  testPropTypesWithErrors(
```
Comment (bujars) [inline __tests__/CommentsTouchBox-test.js:55]: We can also check for {} and []

File: __tests__/ProfileAvatar-test.js [hunk 8]
```diff
@@ -1,10 +1,84 @@
...
+      {username: username},
+      'prop',
+      ProfileAvatar.username,
+    );
+    assert(result === errorMsg);
+  });
+}
+
+describe('testing ProfileAvatar values, should throw errors', () => {
+  testPropTypesWithErrors(
```
Comment (bujars) [inline __tests__/ProfileAvatar-test.js:54]: Should we also require omg to be a string?

Comment (bujars) [reply __tests__/ProfileAvatar-test.js:54]: Ie add prototype tests for img
